### PR TITLE
feat: add eip3074 integration test

### PR DESCRIPTION
**Pull request by fgimenez**
Adds an integration test for the EIP-3074 instructions:
* an AlphaNet node is created
* two smart contracts are deployed:
  * `SenderRecorder` has a function to record msg.sender in storage
  * `GasSponsorInvoker` allows an EOA to call a separate contract using AUTH and AUTHCALL by receiving a commit and a signature from a separate authority EOA
* an EOA without funds signs a commit
* a separate EOA with funds interacts with `GasSponsorInvoker` to call the recorder function in `SenderRecorder` on behalf of the authority 
* the test verifies that the authority was recorded as sender in `SenderRecorder`

As part of the test development a few fixes were required in the instructions implementation:
* The nonce used for the digest should be from the authority
* AUTHCALL should read value_ext and verify that it is 0
* All the gas should be forwarded to the subsequent call if gas_limit == 0

### CODE REVIEW

File: crates/testing/src/tests.rs
Comment (mattsse) [inline crates/testing/src/tests.rs:1]: this is pretty cool!

**Review by mattsse — APPROVED** [review]
nice!
love the test!

pending @Rjected

File: crates/testing/src/tests.rs
Comment (Rjected) [reply crates/testing/src/tests.rs:1]: yeah this is great!

**Review by Rjected — APPROVED** [review]
nice! this is really clean, great work

**Pull request merged**